packages/fx-core/src/common/local/projectSettingsHelper.ts

```diff
@@ -41,8 +42,9 @@ export class ProjectSettingsHelper {
 
   public static includeAAD = (projectSettings: ProjectSettings | undefined): boolean =>
     !ProjectSettingsHelper.isMigrateFromV1(projectSettings) &&
-    (projectSettings?.solutionSettings as AzureSolutionSettings)?.hostType ===
-      HostTypeOptionAzure.id;
+    (projectSettings?.solutionSettings as AzureSolutionSettings)?.activeResourcePlugins?.includes(
```

Comment:
`ProjectSettingsHelper` follows the same logic as plugin activate.
I note that AAD plugin has comment "[// For scaffold, activeResourecPlugins is undefined](https://github.com/OfficeDev/TeamsFx/blob/dev/packages/fx-core/src/plugins/resource/aad/index.ts#L40)", so does here need the same logic? `ProjectSettingsHelper` is used by scaffold as well.

Reply:
The comment in AAD plugin means that when activate() is called in scaffold, activeResourcePlugins is undefined. In this case, when ProjectSettingsHelper is called, activeResourcePlugins already have values.

Reply:
I see, make sense to me.